packages/arcgis-rest-portal/src/items/helpers.ts

```diff
@@ -238,15 +238,20 @@ export function serializeItem(item: IItemAdd | IItemUpdate | IItem): any {
 }
 
 /**
- * requestOptions.owner is given priority, requestOptions.item.owner will be checked next. If neither are present, authentication.username will be assumed.
+ * `requestOptions.owner` is given priority, `requestOptions.item.owner` will be checked next. If neither are present, `authentication.getUserName()` will be used instead.
  */
-export function determineOwner(requestOptions: any): string {
+export function determineOwner(requestOptions: any): Promise<string> {
   if (requestOptions.owner) {
-    return requestOptions.owner;
-  }
-  if (requestOptions.item && requestOptions.item.owner) {
-    return requestOptions.item.owner;
+    return Promise.resolve(requestOptions.owner);
+  } else if (requestOptions.item && requestOptions.item.owner) {
+    return Promise.resolve(requestOptions.item.owner);
+  } else if (requestOptions.authentication) {
```

Comment:
This will work for `UserSession`, but not for `ApplicationSession`, right?
```suggestion
  } else if (requestOptions.authentication && requestOptions.authentication.getUsername) {
```